scripts/check_unit.sh

```diff
@@ -19,8 +19,55 @@ if [ -f profile.out ]; then
 fi
 }
 
+# docker rm returns 1 if the image isn't found. This is OK and expected, so we suppress it
+# Any return status other than 0 or 1 is unusual and so we exit
+remove_docker_container () {
+DOCKER_KILL_EXIT_CODE=0
+docker kill CouchDBStoreTest >/dev/null 2>&1 || DOCKER_KILL_EXIT_CODE=$?
+
+if [ $DOCKER_KILL_EXIT_CODE -ne 0 ] && [ $DOCKER_KILL_EXIT_CODE -ne 1 ]; then
+  echo "error: 'docker kill CouchDBStoreTest' returned ${DOCKER_KILL_EXIT_CODE}, but either 0 or 1 was expected."
+
+  # There's no easy way to print the error message on the screen without temporary files,
+  # so we ask the user to check manually
+  echo "Try running 'docker kill CouchDBStoreTest' manually to see the full error message."
+
+  exit 1
+fi
+
+DOCKER_RM_EXIT_CODE=0
```

Comment:
duplicate code

Reply:
I refactored the error code check into a separate function. Take a look and let me know what you think.